lib/system/file.go

```diff
@@ -77,27 +77,40 @@ func copyAll(src, dst string, fileCount *uint) (err error) {
 	dst = filepath.Clean(dst)
 
 	log.Debugf("copy %s -> %s", src, dst)
+
+	// fetch source permissions
 	si, err := os.Stat(src)
 	if err != nil {
 		return trace.ConvertSystemError(err)
 	}
+
+	// files often don't have execute, but directories need it
+	mode := si.Mode()
 	if si.Mode().IsRegular() {
-		*fileCount++
-		return CopyFile(src, filepath.Join(dst, filepath.Base(src)))
+		mode = mode | os.ModeDir
+		mode = mode | 0100
 	}
 
+	// ensure destination exists
 	_, err = os.Stat(dst)
```

Comment:
Is it necessary to check file info before attempting to make the directory? Will `Stat` return a different error than `MkdirAll`?